ecommerce_worker/sailthru/tasks.py

```
@@ -0,0 +1,248 @@
+                                     _get_configuration('SAILTHRU_SECRET', site_code=site_code))
+
+    # update the "unenrolled" course array in the user record on Sailthru if new enroll or unenroll
+    if new_enroll:
+        if not _update_unenrolled_list(sailthru_client, email, course_url, False):
+            _schedule_retry(self, site_code=site_code)
+
+    # Get course data from Sailthru content library or cache
+    course_data = _get_course_content(course_url, sailthru_client, site_code=site_code)
+
+    # build item description
+    item = _build_purchase_item(course_id, course_url, unit_cost * 100, mode, course_data)
+
+    # build purchase api options list
+    options = {}
+    if incomplete and _get_configuration('SAILTHRU_ABANDONED_CART_TEMPLATE'):
+        options['reminder_template'] = _get_configuration('SAILTHRU_ABANDONED_CART_TEMPLATE', site_code=site_code)
+        options['reminder_time'] = "+{} minutes".format(_get_configuration('SAILTHRU_ABANDONED_CART_DELAY',
+                                                                           site_code=site_code))
+
+    # add appropriate send template
+    if send_template:
+        options['send_template'] = send_template
+
+    if not _record_purchase(sailthru_client, email, item, incomplete, message_id, options):
+        _schedule_retry(self, site_code=site_code)
+
+
+def _schedule_retry(self, site_code=None):
+    """
+    Schedule a retry
+    :param self:
+    :return: Doesn't return
+    """
+    raise self.retry(countdown=_get_configuration('SAILTHRU_RETRY_SECONDS', site_code=site_code),
+                     max_retries=_get_configuration('SAILTHRU_RETRY_ATTEMPTS', site_code=site_code))
+
+
+def _get_configuration(option, site_code=None):
+    """
+    Get a configuration returning None if not found
+    :param option:
+    :return:
+    """
+    try:
+        return get_configuration(option, site_code=site_code)
+    except RuntimeError:
+        return None
```

I think it's dangerous for the app to fail silently when configuration is missing. If something the app needs to operate is missing, I would prefer that it fail hard.